### PR TITLE
Fix: Add user-friendly error messages for Stripe payments

### LINKED ISSUE #6786: Failed Stripe payments should provide user friendly context for errors.
## Details

As a site admin and/or donor, I want a more meaningful error response from Stripe that will allow me to understand why a payment was declined.

Currently, the error messages displayed on the form after a failed Stripe payment give no indication of what when wrong.
For example:
Error on GiveWP:
"The stripe gateway returned an error while processing the donation"
Error in Stripe:
"The bank returned the decline code insufficient_funds. Learn more about declines."

Error on GiveWP
"Error: Error creating payment intent with Stripe. Please try again."
Error in Stripe:
"message": "Your card was declined.",

## Expected Behavior
Errors should provide more context for users as to why the donation was unable to process.

## Steps to Reproduce

1. Create a form and enable Stripe as a payment gateway.
2. Make a donation that will fail because of errors due to the card payment.
3. View Error messages.

## Visuals
Screen Shot 2023-04-27 at 9 56 29 AM

## Additional Context

Original Issue: https://github.com/impress-org/give-stripe/issues/438

(https://secure.helpscout.net/conversation/1131736526/62020?folderId=672194)

## Acceptance Criteria

-   [x] When a payment intent error occurs, the user should receive an error message that provides context on why the payment failed.

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
Resolves #6786

## Description
This pull request adds more user-friendly error messages for Stripe payments. When a user faces an error during the payment process, the new error messages provide more context about the issue, which will help them to take appropriate actions to resolve the issue.

## Affects

Stripe Payments

## Visuals
Screen Shot 2023-05-07 at 10 41 11 AM

## Testing Instructions

- Create a form and setup Stripe as a payment option.
- Make a donation that would generally cause a user-facing error. For example, you can use a test card with no balance. The card number is 4000 0000 0000 9995.
- View new error messages.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: includes/gateways/stripe/includes/class-give-stripe-payment-intent.php [hunk 1]
```diff
@@ -56,21 +57,23 @@ public function create( $args ) {
...
+                    sprintf(
+                    /* translators: %s Exception Error Message */
+                        __('Unable to create a payment intent. Details: %s', 'give'),
+                        $e->getMessage()
+                    )
+                );
+
+                give_set_error('stripe_payment_intent_error',
+                    $e->getMessage()
+                );
```
Comment (JasonTheAdams) [inline includes/gateways/stripe/includes/class-give-stripe-payment-intent.php:72]: I don't really recommend doing this generically with exception messages. I can see below that you're writing the message in the exception itself and displaying it here, but this `catch (Exception $e)` will catch _any_ exception, and then we're risking displaying very technical and confusing messages to the donor.

In general, I try to make exception messages clear to the developer who stumbles on it to help them debug the issue. So I'd probably leave the exception message as it is, and update the message here.

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
Definitely improved messaging, @JoshuaHungDinh! Let's tweak how its done a little bit.

**Review by Benunc — CHANGES_REQUESTED** [review]
Is it normal to break the string into multiple lines? Either way, I added different wording. For posterity:

1. We don't apologize, as it doesn't lead toward resolution.
2. We don't recommend to donors to reach out to GiveWP's support team.
3. We try to avoid the word "payment"

File: src/PaymentGateways/Gateways/Stripe/Traits/HandlePaymentIntentStatus.php [hunk 2]
```diff
@@ -32,7 +33,10 @@ public function handlePaymentIntentStatus(PaymentIntent $paymentIntent, Donation
                 return new PaymentProcessing($paymentIntent->id());
             default:
                 throw new PaymentIntentException(
-                    sprintf(__('Unhandled payment intent status: %s', 'give'), $paymentIntent->status())
+                    sprintf(__('%s Sorry, there was an issue with your payment. Please check your payment method
```
[the remark this answers is not in the document]
Comment (JoshuaHungDinh) [reply src/PaymentGateways/Gateways/Stripe/Traits/HandlePaymentIntentStatus.php]: Resoloved: https://github.com/impress-org/givewp/pull/6787/commits/c7b9ef284c1aaffa97dc503f5b75fd7c8d3a5e0c

File: src/PaymentGateways/Gateways/Stripe/Traits/HandlePaymentIntentStatus.php [hunk 3]
```diff
@@ -32,7 +33,10 @@ public function handlePaymentIntentStatus(PaymentIntent $paymentIntent, Donation
...
+                    or contact your card issuer for assistance. If the issue persists, please try again later or contact our support
```
[the remark this answers is not in the document]
Comment (JoshuaHungDinh) [reply src/PaymentGateways/Gateways/Stripe/Traits/HandlePaymentIntentStatus.php]: Resolved: https://github.com/impress-org/givewp/pull/6787/commits/c7b9ef284c1aaffa97dc503f5b75fd7c8d3a5e0c

File: src/PaymentGateways/Gateways/Stripe/Traits/HandlePaymentIntentStatus.php [hunk 4]
```diff
@@ -32,7 +33,10 @@ public function handlePaymentIntentStatus(PaymentIntent $paymentIntent, Donation
...
+                    team for further assistance.', 'give'),
```
[the remark this answers is not in the document]
Comment (JoshuaHungDinh) [reply src/PaymentGateways/Gateways/Stripe/Traits/HandlePaymentIntentStatus.php]: Resolved: https://github.com/impress-org/givewp/pull/6787/commits/c7b9ef284c1aaffa97dc503f5b75fd7c8d3a5e0c

File: includes/gateways/stripe/includes/class-give-stripe-payment-intent.php (hunk 1 above)
Comment (JoshuaHungDinh) [reply includes/gateways/stripe/includes/class-give-stripe-payment-intent.php:72]: Resolved: https://github.com/impress-org/givewp/pull/6787/commits/c7b9ef284c1aaffa97dc503f5b75fd7c8d3a5e0c

**Review by JasonTheAdams — APPROVED** [review]
Thanks for this improvement, @JoshuaHungDinh!

**Review by Benunc — APPROVED** [review]
looks good!

**Pull request merged**